### PR TITLE
PMM-7568 MongoDB backup restores

### PR DESCRIPTION
**Pull request by artemgavrilov**
[PMM-7568](https://jira.percona.com/browse/PMM-7568)

- [ ] Build: https://github.com/Percona-Lab/pmm-submodules/pull/1836
---
- [x] https://github.com/percona/pmm/pull/742
- [x] https://github.com/percona/pmm-agent/pull/273
---
- [ ] Tests passed.
- [ ] Feature build pass.
- [ ] (Re)requested review.
- [ ] Fix conflicts with target branch.
- [ ] Update API dependency.
- [ ] Update jira ticket description if needed.
- [ ] Attach screenshots/console output to confirm new behavior to jira ticket, if applicable.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -34,7 +34,7 @@ require (
 	github.com/minio/minio-go/v7 v7.0.10
 	github.com/percona-platform/dbaas-api v0.0.0-20210518140815-21d371355e30
 	github.com/percona-platform/saas v0.0.0-20210505183502-c18b6f47c932
-	github.com/percona/pmm v0.0.0-20210601163123-a8b03f81a484
+	github.com/percona/pmm v0.0.0-20210604182633-264e7f18014e
```
Comment (artemgavrilov) [inline go.mod:37]: TODO

**Review by Dasio — APPROVED** [review]

File: services/agents/jobs.go [hunk 2]
```diff
@@ -218,6 +226,67 @@ func (s *JobsService) StartMySQLRestoreBackupJob(
...
+		Name:     name,
+		User:     dbConfig.User,
+		Password: dbConfig.Password,
+		Address:  dbConfig.Address,
+		Port:     int32(dbConfig.Port),
+		Socket:   dbConfig.Socket,
+	}
+
+	switch {
+	case locationConfig.S3Config != nil:
```
Comment (oter) [inline services/agents/jobs.go:252]: you already have a check at line 238

File: services/management/backup/backups_service.go [hunk 3]
```diff
@@ -159,11 +156,36 @@ func (s *BackupsService) RestoreBackup(
...
+			jobType = models.MongoDBRestoreBackupJob
+			jobResultData = &models.JobResultData{
+				MongoDBRestoreBackup: &models.MongoDBRestoreBackupJobResult{
+					RestoreID: restoreID,
+				}}
+		case models.PostgreSQLServiceType,
+			models.ProxySQLServiceType,
+			models.HAProxyServiceType,
+			models.ExternalServiceType:
+			return errors.Errorf("backup restore unimplemeted for : %s", service.ServiceType)
```
Comment (oter) [inline services/management/backup/backups_service.go:183]:
```suggestion
			return errors.Errorf("backup restore unimplemented for service type: %s", service.ServiceType)
```

**Review by oter — APPROVED** [review]

File: services/management/backup/backups_service.go (hunk 3 above)
Comment (artemgavrilov) [reply services/management/backup/backups_service.go]: Done

File: services/agents/jobs.go (hunk 2 above)
Comment (artemgavrilov) [reply services/agents/jobs.go:248]: Yep, removed the redundant check.

File: services/agents/jobs.go [hunk 4]
```diff
@@ -218,6 +226,63 @@ func (s *JobsService) StartMySQLRestoreBackupJob(
...
+		mongoDBReq.LocationConfig = &agentpb.StartJobRequest_MongoDBRestoreBackup_S3Config{
+			S3Config: &agentpb.S3LocationConfig{
+				Endpoint:     locationConfig.S3Config.Endpoint,
+				AccessKey:    locationConfig.S3Config.AccessKey,
+				SecretKey:    locationConfig.S3Config.SecretKey,
+				BucketName:   locationConfig.S3Config.BucketName,
+				BucketRegion: locationConfig.S3Config.BucketRegion,
+			},
```
Comment (BupycHuk) [inline services/agents/jobs.go:256]: looks like it can be extracted.

**Review by BupycHuk — APPROVED** [review]

File: services/agents/jobs.go (hunk 4 above)
Comment (artemgavrilov) [reply services/agents/jobs.go]: Done

**Pull request merged**